hackathon/mdakits/projects.md

```diff
@@ -0,0 +1,182 @@
+[clustering]: https://en.wikipedia.org/wiki/Cluster_analysis
+
+[scikit-learn clustering]: https://scikit-learn.org/stable/modules/clustering.html
+
+[UserGuide AnalysisBase]: https://userguide.mdanalysis.org/stable/examples/analysis/custom_trajectory_analysis.html
+
+## Clustered density analysis ##
+
+**TODO**
+
+Clustered density analysis: (1) cluster trajectory, (2) create
+densities for each cluster.
+
+
+## New file formats ##
+
+**TODO**
+
+New file format as a MDAKit: 
+
+- pdbx or mmcif (https://github.com/MDAnalysis/mdanalysis/issues/2367)
+— can be initially rough and just use any other dependencies to get
+  the job done and we can then later see if we want to move it to
+  core.
+  
+  
+
+## Solvent accessible surface (SASA) ##
+
+
+## Molecular volume and surface analysis
+
+*This is one of the most wanted analysis features missing in MDAnalysis.*
+
+It is often necessary to measure volume and surface area of a
+biomolecule or parts of it over a MD trajectory. MDAnalysis is
+currently lacking this important functionality. In this project you
+will implement an analysis class that calculates the molecular volume
+and area for an atomgroup as a function of time. See issue
+[#2439](https://github.com/MDAnalysis/mdanalysis/issues/2439).
+
+The [FreeSASA](http://freesasa.github.io/) library appears to be a
+suitable tool to integrate into MDAnalysis. It comes under MIT license
+and has a C core and python bindings:
+* C-library: https://github.com/mittinatten/freesasa
+* Python bindings: https://github.com/freesasa/freesasa-python
+
+> By default Lee & Richards' algorithm is used, but Shrake & Rupley's
+> is also available.
+
+
+
+### Objectives
+
+For this project you would
+
+1. Interface [freesasa-python](https://github.com/freesasa/freesasa-python)
+   [[Mitternacht 2016]](#Mitternacht2016) with MDAnalysis: Find out
+   how to pass information from MDAnalysis to `freesasa` so that it
+   can compute areas and volumes. 
+2. Write a `SASAnalysis` class using the `AnalysisBase` framework (see
+   the [tutorial on writing your own trajectory analysis][[UserGuide
+   AnalysisBase]].
+3. create test cases (use existing files in MDA and run external
+   implementation for reference)
+4. benchmark performance
+
+
+### References ###
+
+1. <a id="Mitternacht2016"/>Simon Mitternacht (2016) FreeSASA: An open
+   source C library for solvent accessible surface area
+   calculation. F1000Research 5:189 (doi:
+   [10.12688/f1000research.7931.1](https://doi.org/10.12688/f1000research.7931.1))
+
+
+## Contact analysis ##
+
+**TODO**
+
+Contact analysis (however you define it but really just something
+better than native contacts)
+
+## Structural analysis ##
+
+**TODO**
+
+- DSSP
+- stride
+- ... ?
+
+
+## Voronoi tessellation ##
+
+**TODO**
+
+## Ring puckering ##
+
+**TODO**
+
+Ring puckering 
+
+- Cremer-Pople for 6+ membered rings
+- Hill Reilley, Pickett angles
+- ...
+
+## Converters for other libraries ##
```

Comment:
Can we keep converters out of the list of things for now? (as per recent conversations)

Reply:
sure